perimeter/docs/resource_policy_shared_accounts_access.md

```diff
@@ -0,0 +1,68 @@
+This benchmark answers the following questions:
+
+- What resources have resource policies that allow untrusted cross account access?
+
+This benchmark defines shared as a policy having at least one `Allow` statement that grants one or more permissions to trusted accounts, e.g.,
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "AllowPublicAccess1",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "111122223333"
+      },
+      "Action": ["s3:PutObject", "s3:PutObjectAcl"],
+      "Resource": "arn:aws:s3:::EXAMPLE-BUCKET/*"
+    }
+  ]
+}
+```
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "AllowPublicAccess2",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::111122223333:root"
+      },
+      "Action": ["s3:PutObject", "s3:PutObjectAcl"],
+      "Resource": "arn:aws:s3:::EXAMPLE-BUCKET/*"
+    }
+  ]
+}
+```
+
+When evaluating statements for shared access, the following [condition keys](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html) are checked:
+
+- `aws:PrincipalAccount`
+- `aws:PrincipalArn`
+- `aws:PrincipalOrgID`
+- `aws:SourceAccount`
+- `aws:SourceArn`
+- `aws:SourceOwner`
+
+And the following [condition operators](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html) are checked:
+
+- `ArnLike`
+- `ArnEquals`
+- `StringEquals`
+- `StringEqualsIgnoreCase`
+- `StringLike`
+
+For each statement, if there are any condition keys then these condition keys will be evaluated as follows:
+
+Principals conditions are checked against the Policy Principals.
+If Principals conditions have smaller scope that the Policy Principals then the analyzer will reduce the scopeage.
+If Principals conditions have larger scope that the Policy Principals then the analyzer will leave the Policy Principals unchanged.
+If Principals conditions have a scope that doesn't contain the the Policy Principals then the analyzer will return this as invalid.
```

Comment:
```suggestion
For each statement, if there are any condition keys then these condition keys will be evaluated as follows:
- Principals conditions are checked against the Policy Principals.
- If Principals conditions have smaller scope that the Policy Principals then the analyzer will reduce the scopeage.
- If Principals conditions have larger scope that the Policy Principals then the analyzer will leave the Policy Principals unchanged.
- If Principals conditions have a scope that doesn't contain the the Policy Principals then the analyzer will return this as invalid.
```

Beside making this into a proper Markdown list, a few questions/comments:
- Should we phrase conditions as `principals' conditions`, or `statement's conditions`? When I think about principals and conditions, I associate them with the statement they're in (since our evaluation starts with per statement, and then combines these results for the entire policy). The current wording confused me a bit since a given principal in a policy statement doesn't have a specific condition assigned, but instead, the condition is applied to the whole statement.
- For this line: `If Principals conditions have a scope that doesn't contain the the Policy Principals then the analyzer will return this as invalid.`, do you have an example you can add to this comment showing this? What does the table return if I have a statement that grants access to account `012345678912`, and a condition that limits access to source account `456789012345`?